### PR TITLE
Fix Helm Version Check

### PR DESCRIPTION
**Pull request by sujuka99**

### CODE REVIEW

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 1]
```diff
@@ -457,17 +457,22 @@ def test_should_call_run_command_method_when_helm_template_without_optional_args
         )
 
     @pytest.mark.parametrize(
-        "version",
+        "version_object, version",
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:460]: consider swapping those (as input, output in that order) and give them names where it's more clear what they're used for, e.g. `raw_version`, `expected_version` or similar

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 2]
```diff
@@ -457,17 +457,22 @@ def test_should_call_run_command_method_when_helm_template_without_optional_args
...
         [
-            "v3.12.0+gc9f554d",
-            "v3.12.0",
-            "v3.12",
-            "v3",
+            (Version(major=3, minor=12, patch=0), "v3.12.0+gc9f554d"),
+            (Version(major=3, minor=12, patch=0), "v3.12.0"),
+            (Version(major=3, minor=12, patch=0), "v3.12"),
+            (Version(major=3, minor=0, patch=0), "v3"),
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:465]: consider dropping the argument names

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py [hunk 3]
```diff
@@ -457,17 +457,22 @@ def test_should_call_run_command_method_when_helm_template_without_optional_args
...
         ],
     )
-    def test_should_call_helm_version(self, run_command: MagicMock, version):
+    def test_should_call_helm_version(
+        self,
+        run_command: MagicMock,
+        version_object,
```
Comment (disrupted) [inline tests/component_handlers/helm_wrapper/test_helm_wrapper.py:471]:
```suggestion
        version_object: Version,
```

**Review by disrupted — CHANGES_REQUESTED** [review]

File: kpops/component_handlers/helm_wrapper/helm.py [hunk 4]
```diff
@@ -198,14 +198,11 @@ def get_manifest(self, release_name: str, namespace: str) -> Iterable[HelmTempla
...
-        version_match = re.search(r"^v(\d+(\.\d+){0,2})", short_version)
+        version_match = re.search(r"^v(\d+(?:\.\d+){0,2})", short_version)
         if version_match is None:
             raise RuntimeError("Could not parse the Helm version.")
-        version = version_match.group(1)
-        while re.match(r"^(\d+(\.\d+)?)$", version):
-            version += ".0"
-        major, minor, patch = map(int, version.split("."))
-        return Version(major, minor, patch)
+        version = map(int, version_match.group(1).split("."))
```
Comment (disrupted) [inline kpops/component_handlers/helm_wrapper/helm.py:204]:
```suggestion
        parsed_version = map(int, version_match.group(1).split("."))
```
wdyt?

File: kpops/component_handlers/helm_wrapper/helm.py [hunk 5]
```diff
@@ -198,14 +198,11 @@ def get_manifest(self, release_name: str, namespace: str) -> Iterable[HelmTempla
     def get_version(self) -> Version:
         command = ["helm", "version", "--short"]
         short_version = self.__execute(command)
-        version_match = re.search(r"^v(\d+(\.\d+){0,2})", short_version)
+        version_match = re.search(r"^v(\d+(?:\.\d+){0,2})", short_version)
         if version_match is None:
             raise RuntimeError("Could not parse the Helm version.")
```
Comment (disrupted) [inline kpops/component_handlers/helm_wrapper/helm.py:203]: should we add the output of Helm to the error message?

File: kpops/component_handlers/helm_wrapper/helm.py (hunk 4 above)
Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py:204]: looks good, will do

File: kpops/component_handlers/helm_wrapper/helm.py (hunk 5 above)
Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py:203]: sounds useful, I would do it

Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py]: wdyt?

File: tests/component_handlers/helm_wrapper/test_helm_wrapper.py (hunk 1 above)
Comment (sujuka99) [reply tests/component_handlers/helm_wrapper/test_helm_wrapper.py]: done

File: kpops/component_handlers/helm_wrapper/helm.py [hunk 6]
```diff
@@ -200,11 +200,13 @@ def get_version(self) -> Version:
...
-        while re.match(r"^(\d+(\.\d+)?)$", version):
-            version += ".0"
-        major, minor, patch = map(int, version.split("."))
+            raise RuntimeError(
+                f"Could not parse the Helm version.\n\nHelm output:\n{short_version}"
+            )
+        parsed_version = version_match.group(1)
+        while re.match(r"^(\d+(\.\d+)?)$", parsed_version):
+            parsed_version += ".0"
+        major, minor, patch = map(int, parsed_version.split("."))
```
Comment (disrupted) [inline kpops/component_handlers/helm_wrapper/helm.py:209]: this seems to have been reverted

Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py:209]: ???

Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py:209]: Will fix it

File: kpops/component_handlers/helm_wrapper/helm.py (hunk 5 above)
Comment (disrupted) [reply kpops/component_handlers/helm_wrapper/helm.py]: looks good

File: kpops/component_handlers/helm_wrapper/helm.py (hunk 6 above)
Comment (sujuka99) [reply kpops/component_handlers/helm_wrapper/helm.py]: Must be ok now.

**Review by disrupted — APPROVED** [review]

**Pull request merged**